src/components/Modals/WorldPermissionsModal/ModelTabs/Layouts/WorldPermissionsAddUserForm/WorldPermissionsAddUserForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { Icon } from 'decentraland-ui'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+
+import { Field } from 'decentraland-ui/dist/components/Field/Field'
+import { WorldPermissionNames } from 'lib/api/worlds'
+
+import styles from './WorldPermissionsAddUserForm.module.css'
+import { Props } from './WorldPermissionsAddUserForm.types'
+
+export const WORLD_PERMISSIONS_ADD_USER_FORM_SHOW_FORM_BUTTON_DATA_TEST_ID = 'world-permissions-add-user-form-show-form-button-data-test-id'
+export const WORLD_PERMISSIONS_ADD_USER_FORM_FIELD_DATA_TEST_ID = 'world-permissions-add-user-form-field-data-test-id'
+export const WORLD_PERMISSIONS_ADD_USER_FORM_CHANGE_PERMISSION_BUTTON_DATA_TEST_ID =
+  'world-permissions-add-user-form-change-permission-button-data-test-id'
+
+export const WorldPermissionsAddUserForm = React.memo((props: Props) => {
+  const {
+    showAddUserForm,
+    newAddress,
+    isLoadingNewUser,
+    addButtonLabel,
+    error,
+    onShowAddUserForm,
+    onNewAddressChange,
+    onUserPermissionListChange
+  } = props
+
+  return (
+    <div className={styles.addUserWrapper}>
+      {!showAddUserForm ? (
+        <div className={styles.addUserButtonContainer}>
+          <Button data-testid={WORLD_PERMISSIONS_ADD_USER_FORM_SHOW_FORM_BUTTON_DATA_TEST_ID} onClick={onShowAddUserForm}>
+            <Icon name="plus" />
+            {addButtonLabel}
+          </Button>
+        </div>
+      ) : (
+        <div className={styles.addUserFormContainer}>
+          <Field
```

Comment:
Please, check the `AddressField` component (from the decentraland-dapps repo) that should be more suitable for this task. It does some address checking and it has an auto-name resolver.